### ISSUE TITLE
How do I save Image using LZW compression?

### ISSUE THREAD

**Issue by artzers** [blink1073/tifffile#20]
Hello, I am using Windows Python3.6. I want to read, process some 3D tiff images and save them using LZW compression. I failed to install python libtiff, success installing tifffile. tifffile can read my 3D tiff images well. However I am wondering how to save tiff image using LZW. 

I find the source code:
elif compress:
            def compress(data, level=compress):
                return zlib.compress(data, level)

tifffile seems just using zlib compression.
Please help me, thanks

**Comment by cgohlke** [comment]
Tifffile supports writing with [Zip or LZMA](https://github.com/blink1073/tifffile/blob/master/tifffile/tifffile.py#L537) compression. Zip and LZMA are available in the Python standard library and are more effective than LZW, which can even result in files that are larger than uncompressed files.

**Comment by artzers** [comment]
@cgohlke      We used to use LZW to save lossless compression tiff image. Losslessness is very important. So how to compress tiff image lossless using Zip or LZMA? For example , Zip compression require input value. 0 is noncompression and I do not need this. I wonder if I set 3 or 4 for Zip, whether the result tiff image is still lossless? Thank you.

**Comment by cgohlke** [comment]
Zip and LZMA are lossless compression techniques.

**Comment by cgohlke** [comment]
Also note that tifffile does not use predictors and compresses one stripe per plane by default, which can be very inefficient depending on your data. To compress smaller chunks of data use tiles. If you need access to all available options of the TIFF specification, better use the libtiff library.

**Comment by artzers** [comment]
@cgohlke  So it means it is still lossless if I set value 2~9 for Zip compression?
Well~~Otherwise it seems that there is no one developing libtiff for  python 3.5 up to now. Thank you.

**Comment by MeowMeowLady** [comment]
Hi, did you solve the problem? I failed using lzw compression mode with tifffile too.

**Comment by artzers** [comment]
@MeowMeowLady No, I just use tifffile uncompress mode. There are no other python package which conclude LZW part.

**Comment by user-na** [comment]
@MeowMeowLady, @artzers I just faced the same question. I will share my insights here in case anyone with a similar question finds your post, like I did.
While looking at the `save ` function I realized it does have a compatibility to LZMA. If you feed `compress = 'LZMA'` to the `imsave` function it will feed it to the writer, resulting in compressed tiffs. The `compress` keyword has the following description:
```
compress : int or 'LZMA', 'ZSTD'
            Values from 0 to 9 controlling the level of zlib compression.
            If 0 (default), data are written uncompressed.
            Compression cannot be used to write contiguous files.
            If 'LZMA' or 'ZSTD', LZMA or ZSTD compression is used, which is
            not available on all platforms.
```
